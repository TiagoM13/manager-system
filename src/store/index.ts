export * from './modules/dialog';
export * from './modules/menu';
export * from './modules/image';
export * from './modules/auth';
export * from './modules/name';
export * from './modules/profile';
export * from './modules/patient';
