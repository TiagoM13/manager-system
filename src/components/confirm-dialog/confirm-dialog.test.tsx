import React from 'react';

import { useDialog } from '@/store';
import { fireEvent, render } from '@testing-library/react';

import { ConfirmDialog } from '.';

jest.mock('@/store', () => ({
  useDialog: jest.fn(),
}));

describe('>ConfirmDialog/>', () => {
  const mockCloseDialog = jest.fn();
  const mockAccept = jest.fn();

  const defaultConfig = {
    header: 'Confirmação',
    message: 'Você tem certeza?',
    acceptLabel: 'Sim',
    rejectLabel: 'Não',
    accept: mockAccept,
    rejectBtnColor: 'secondary',
    acceptBtnColor: 'danger',
    acceptIcon: null,
    rejectIcon: null,
  };

  beforeEach(() => {
    jest.clearAllMocks();

    (useDialog as unknown as jest.Mock).mockReturnValue({
      isOpenDialog: true,
      closeDialog: mockCloseDialog,
      config: defaultConfig,
    });
  });

  test('should render correctly component', () => {
    const { getByText, getByRole, asFragment } = render(<ConfirmDialog />);

    expect(getByText('Confirmação')).toBeInTheDocument();
    expect(getByText('Você tem certeza?')).toBeInTheDocument();
    expect(getByText('Sim')).toBeInTheDocument();
    expect(getByText('Não')).toBeInTheDocument();

    expect(getByRole('button', { name: /sim/i })).toBeInTheDocument();
    expect(getByRole('button', { name: /não/i })).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  test('should trigger accept function and close dialog when clicking "Sim"', () => {
    const { getByRole } = render(<ConfirmDialog />);

    fireEvent.click(getByRole('button', { name: /sim/i }));

    expect(mockAccept).toHaveBeenCalled();
    expect(mockCloseDialog).toHaveBeenCalled();
  });

  test('should close dialog when clicking "Não"', () => {
    const { getByRole } = render(<ConfirmDialog />);

    fireEvent.click(getByRole('button', { name: /não/i }));

    expect(mockCloseDialog).toHaveBeenCalled();
    expect(mockAccept).not.toHaveBeenCalled();
  });
});
