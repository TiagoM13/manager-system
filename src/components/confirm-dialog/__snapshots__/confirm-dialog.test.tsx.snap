// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>ConfirmDialog/> should render correctly component 1`] = `
<DocumentFragment>
  <div
    aria-labelledby="modal-title"
    aria-modal="true"
    class="fixed inset-0 z-[100] flex items-center justify-center bg-black bg-opacity-50"
    role="dialog"
  >
    <div
      class="bg-white rounded-lg max-w-lg shadow-lg w-full p-4"
    >
      <div
        class="flex justify-between items-center mb-2"
      >
        <h2
          class="text-xl font-semibold"
          id="modal-title"
        />
        <button
          aria-label="Close modal"
          class="text-slate-600 rounded-full p-[0.125rem] hover:bg-slate-200"
          data-testid="btn-close-dialog"
        >
          <svg
            class="size-5"
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M208.49,191.51a12,12,0,0,1-17,17L128,145,64.49,208.49a12,12,0,0,1-17-17L111,128,47.51,64.49a12,12,0,0,1,17-17L128,111l63.51-63.52a12,12,0,0,1,17,17L145,128Z"
            />
          </svg>
        </button>
      </div>
      <div>
        <div
          class="flex flex-col items-center w-full"
        >
          <div
            class="flex items-center justify-center w-12 h-12 rounded-full bg-red-500/10"
          >
            <svg
              class="size-10 text-red-400"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm0,192a88,88,0,1,1,88-88A88.1,88.1,0,0,1,128,216Zm-8-80V80a8,8,0,0,1,16,0v56a8,8,0,0,1-16,0Zm20,36a12,12,0,1,1-12-12A12,12,0,0,1,140,172Z"
              />
            </svg>
          </div>
          <div
            class="text-center mt-4"
          >
            <h2
              class="text-xl font-semibold"
            >
              Confirmação
            </h2>
            <div>
              Você tem certeza?
            </div>
          </div>
        </div>
        <div
          class="flex justify-end gap-4 mt-10"
        >
          <button
            class="bg-slate-600 outline-slate-500 hover:bg-slate-500 text-white disabled:bg-slate-700 rounded-lg py-2.5 px-3 font-medium text-sm flex gap-2 items-center justify-center w-full"
            data-testid="btn-undefined"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M208.49,191.51a12,12,0,0,1-17,17L128,145,64.49,208.49a12,12,0,0,1-17-17L111,128,47.51,64.49a12,12,0,0,1,17-17L128,111l63.51-63.52a12,12,0,0,1,17,17L145,128Z"
              />
            </svg>
            Não
          </button>
          <button
            class="bg-red-600 outline-red-500 hover:bg-red-500 text-white disabled:bg-red-700 rounded-lg py-2.5 px-3 font-medium text-sm flex gap-2 items-center justify-center w-full"
            data-testid="btn-undefined"
          >
            <svg
              class="size-4"
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M216,48H180V36A28,28,0,0,0,152,8H104A28,28,0,0,0,76,36V48H40a12,12,0,0,0,0,24h4V208a20,20,0,0,0,20,20H192a20,20,0,0,0,20-20V72h4a12,12,0,0,0,0-24ZM100,36a4,4,0,0,1,4-4h48a4,4,0,0,1,4,4V48H100Zm88,168H68V72H188ZM116,104v64a12,12,0,0,1-24,0V104a12,12,0,0,1,24,0Zm48,0v64a12,12,0,0,1-24,0V104a12,12,0,0,1,24,0Z"
              />
            </svg>
            Sim
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
